modules/core/src/copy.cpp

```diff
@@ -93,15 +93,23 @@ copyMask_<uchar>(const uchar* _src, size_t sstep, const uchar* mask, size_t mste
         #if CV_SIMD
         {
             v_uint8 v_zero = vx_setzero_u8();
+            int stride = v_uint8x16::nlanes * 2;
+            int len0 = size.width & -stride;
 
-            for( ; x <= size.width - v_uint8::nlanes; x += v_uint8::nlanes )
+            for( ; x < len0; x += stride )
```

Comment:
Please use straightforward condition: `x <= size.width - v_uint8x16::nlanes * 2`.

`& -stride` trick assumes that initial and all `x` values are multiplier of `stride` and stride is power of 2 (or we should add assertion for that).